src/core.ts

```diff
@@ -163,1775 +120,485 @@ export interface CombineSignature {
-      throw new Error('A MemoryStream was given to imitate(), but it only ' +
-      'supports a Stream. Read more about this restriction here: ' +
-      'https://github.com/staltz/xstream#faq');
-    }
-    this._target = target;
-    for (let ils = this._ils, N = ils.length, i = 0; i < N; i++) {
-      target._add(ils[i]);
-    }
-    this._ils = [];
-  }
-
-  /**
-   * Forces the Stream to emit the given value to its listeners.
-   *
-   * As the name indicates, if you use this, you are most likely doing something
-   * The Wrong Way. Please try to understand the reactive way before using this
-   * method. Use it only when you know what you are doing.
-   *
-   * @param value The "next" value you want to broadcast to all listeners of
-   * this Stream.
-   */
-  shamefullySendNext(value: T) {
-    this._n(value);
-  }
-
-  /**
-   * Forces the Stream to emit the given error to its listeners.
-   *
-   * As the name indicates, if you use this, you are most likely doing something
-   * The Wrong Way. Please try to understand the reactive way before using this
-   * method. Use it only when you know what you are doing.
-   *
-   * @param {any} error The error you want to broadcast to all the listeners of
-   * this Stream.
-   */
-  shamefullySendError(error: any) {
-    this._e(error);
-  }
-
-  /**
-   * Forces the Stream to emit the "completed" event to its listeners.
-   *
-   * As the name indicates, if you use this, you are most likely doing something
-   * The Wrong Way. Please try to understand the reactive way before using this
-   * method. Use it only when you know what you are doing.
-   */
-  shamefullySendComplete() {
-    this._c();
+  unsubscribe (): void {
+    this.dispose()
   }
 }
 
-export class MemoryStream<T> extends Stream<T> {
-  private _v: T;
-  private _has: boolean = false;
-  constructor(producer: InternalProducer<T>) {
-    super(producer);
-  }
-
-  _n(x: T) {
-    this._v = x;
-    this._has = true;
-    super._n(x);
-  }
-
-  _add(il: InternalListener<T>): void {
-    if (this._has) { il._n(this._v); }
-    super._add(il);
-  }
-
-  _lateStop() {
-    this._has = false;
-    super._lateStop();
-  }
-
-  _x(): void {
-    this._has = false;
-    super._x();
-  }
-
-  map<U>(project: (t: T) => U): MemoryStream<U> {
-    return <MemoryStream<U>> this._map(project);
-  }
-
-  mapTo<U>(projectedValue: U): MemoryStream<U> {
-    return <MemoryStream<U>> super.mapTo(projectedValue);
-  }
-
-  take(amount: number): MemoryStream<T> {
-    return <MemoryStream<T>> super.take(amount);
-  }
-
-  endWhen(other: Stream<any>): MemoryStream<T> {
-    return <MemoryStream<T>> super.endWhen(other);
-  }
-
-  replaceError(replace: (err: any) => Stream<T>): MemoryStream<T> {
-    return <MemoryStream<T>> super.replaceError(replace);
-  }
-
-  debug(labelOrSpy?: string | ((t: T) => void)): MemoryStream<T> {
-    return <MemoryStream<T>> super.debug(labelOrSpy);
+export class CompositeSubscription implements Subscription {
```

Comment:
In RxJS v5, their `Subscription` is already a composite subscription. I'd recommend we call our CompositeSubscription just `Sub` (saving space on character count is a thing we do *shrug*).